bot/exts/halloween/spooknamerate.py

```diff
@@ -0,0 +1,392 @@
+                    await channel.send(" ".join(party))
+
+                else:
+                    # no participants in current iteration
+                    await channel.send('Hmm... Looks like no one participated! :cry:')
+
+            async with self.checking_messages:  # Acquire the lock to delete the messages
+                await self.messages.clear()  # reset the messages
+
+        # send the next name
+        self.name = f"{random.choice(self.first_names)} {random.choice(self.last_names)}"
+        await self.data.set('name', self.name)
+
+        await channel.send(
+            "Let's move on to the next name!\nAnd the next name is...\n"
+            f"**{self.name}**!\nTry to spookify that... :smirk:"
+        )
+
+        self.poll = False  # accepting responses
+
+    @announce_name.before_loop
+    async def wait_till_scheduled_time(self) -> None:
+        """Waits till the next day's 12AM if crossed it, otherwise waits till the same day's 12AM."""
+        now = datetime.utcnow()
+        if now.hour < 12:
+            twelve_pm = now.replace(hour=12, minute=0, second=0, microsecond=0)
+            time_left = twelve_pm - now
+            await asyncio.sleep(time_left.seconds)
+            return
+
+        tomorrow_12pm = now + timedelta(days=1)
+        tomorrow_12pm = tomorrow_12pm.replace(hour=12, minute=0, second=0, microsecond=0)
+        await asyncio.sleep((tomorrow_12pm - now).seconds)
+
+    async def get_responses_list(self, final: bool = False) -> Embed:
+        """Returns an embed containing the responses of the people."""
+        channel = await self.get_channel()
+
+        embed = Embed(
+            color=Colour.red()
+        )
+
+        if await self.messages.items():
+            if final:
+                embed.title = "Spook Name Rate is about to end!"
+                embed.description = ("This Spook Name Rate round is about to end in 2 hours! You can review "
+                                     "the entries below! Have you rated other's names?")
+            else:
+                embed.title = "All the spookified names!"
+                embed.description = "See a list of all the entries entered by everyone!"
+        else:
+            embed.title = "No one has added an entry yet..."
+
+        for message_id, data in await self.messages.items():
+            data = json.loads(data)
+
+            embed.add_field(
+                name=(await self.bot.get_user(data['author'])).name,
+                value=f"[{(data)['name']}](https://discord.com/channels/{Client.guild}/{channel.id}/{message_id})",
+            )
+
+        return embed
+
+    async def get_channel(self) -> Union[TextChannel, None]:
+        """Gets the sir-lancebot-channel after waiting until ready."""
+        await self.bot.wait_until_ready()
+        channel = self.bot.get_channel(Channels.community_bot_commands) or await self.bot.fetch_channel(
+            Channels.community_bot_commands
+        )
```

Comment:
Hard to see that this is an or statement, seems a lot more like a very long function call with a parameter.

```suggestion
        channel = (
            self.bot.get_channel(Channels.community_bot_commands)
            or await self.bot.fetch_channel(Channels.community_bot_commands)
        )
```

Reply:
alternatively 
```suggestion
        channel = (self.bot.get_channel(Channels.community_bot_commands) 
                   or await self.bot.fetch_channel(Channels.community_bot_commands))
```
But i'm not sure im a fan of that

Reply:
We don't want that this fail when channel is not in cache.

Reply:
Does this convert it to a bool, by wrapping it in parenthesis? In that case TIL.

Reply:
Both of these return `None` or channel. `None` is false, but channel is true. So when channel exists, this is channel, else `None`.

Reply:
If fail to see the difference then? I simply suggeted wrapping it in parenthesis to linewrap the expression, making it easier to read.

Reply:
I will format it as you suggested in the first part, the second suggestion looks sort of... _malformed_ or something.